crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3655,63 +3651,13 @@ impl<'db> TypeInferenceBuilder<'db> {
                         _ => {}
                     }
                 }
-
-                outcome.return_type(self.db())
             }
-            Err(err) => {
-                // TODO: We currently only report the first error. Ideally, we'd report
-                //  an error saying that the union type can't be called, followed by a sub
-                //  diagnostic explaining why.
-                fn report_call_error(
-                    context: &InferContext,
-                    err: CallError,
-                    call_expression: &ast::ExprCall,
-                ) {
-                    match err {
-                        CallError::NotCallable { not_callable_type } => {
-                            context.report_lint(
-                                &CALL_NON_CALLABLE,
-                                call_expression,
-                                format_args!(
-                                    "Object of type `{}` is not callable",
-                                    not_callable_type.display(context.db())
-                                ),
-                            );
-                        }
```

Comment:
These lints are now handled by `Bindings::report_diagnostics`, which means that all call-site-related lints are now in the same place.